Civi/Api4/Generic/ValidateAction.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Api4\Generic;
+
+use Civi\Api4\Event\ValidateValuesEvent;
+
+/**
+ * Base class for all `Validate` api actions.
+ *
+ * @method $this setValues(array $values) Set all field values from an array of key => value pairs.
+ * @method array getValues() Get field values.
+ *
+ * @package Civi\Api4\Generic
+ */
+class ValidateAction extends AbstractAction {
+
+  use Traits\GetSetValueTrait;
+
+  /**
+   * Run the api Action.
+   *
+   * @param \Civi\Api4\Generic\Result $result
+   *
+   */
+  public function _run(Result $result): void {
+    $e = new ValidateValuesEvent($this, [$this->getValues()], new \CRM_Utils_LazyArray(function () {
+      return [['old' => NULL, 'new' => $this->getValues()]];
+    }));
+
+    $this->onValidateValues($e);
```

Comment:
See my comment https://github.com/civicrm/civicrm-core/pull/30716#discussion_r1696994410 re. implementing this as a subscriber